docs/adr/0045-testing-updates-and-recommendations.md

```diff
@@ -0,0 +1,233 @@
+  - **Bad**, because it's a massive under taking, and the tests already run today without issue.
+  - **Bad**, Don't fix what's not broken
+- Pilot (targeted) Migration of Select Projects **👍 Recommended**
+  - **Good**, because we gain a concrete understanding of the level of effort to migrate from `mocha` -> `jest`; learnings can be captured should we decide to move other projects
+  - **Good**, because we can target high traffic projects where the value would be most felt
+  - **Bad**, because we may end up in the same position
+- Maintain the Status Quo
+  - **Good**, because we don't have to fix what's not borken. Tests are running just fine
+  - **Good**, because focus can be kept on additions and new projects
+  - **Bad**, because we're just pushing off the other problems
+  </details>
+
+<details>
+<summary>B: Optimize Functional Tests</summary>
+
+Options:
+
+1. Audit and Refactor for Shift Left **👍 Recommended**
+2. Audit and Identify Inaccurate/False Tests
+3. Status Quo **👎 Not Recommended**
+
+#### Details
+
+A recent dicovery found that one of our functional tests wasn't actually [testing what we thought it was][invalid test thread]. This lead to discussion about the tests viability and if it would make more sense to split it the functional test into a few different tests at the unit level. This would dramatically speed up the execution, more directly target the functionality (redirect logic and return/retry logic).
+
+It's a good idea to audit functional tests on a regular basis regardless, making sure they're still valid business logic, testing what you expect them to test, and this would give us a chance to find tests that fit a similar bucket, split them apart into unit tests, and speed up test time and coverage. There was a previous [audit][functional test audit] done, and we should revisit this with an eye on what tests can be split up and pushing left, and leveraging more features of Playwright like tagging.
+
+Alternatively, we can just look for the tests that fit the bill, and fix them in place. This wouldn't require as extensive an investment, and would give us more confidence in our functional tests.
+
+#### Pros and Cons of each option
+
+- Audit and Refactor Functional Tests for Shift Left
+  - **Good**, because it's necessary to audit functional tests on a semi-regular basis
+  - **Good**, because we have the potential to reduce the number of functional tests, reducing the amount of time and CI Resources required to execute them.
+  - **Good**, because an audit gives us a clear picture of what tests smoke tests for different environments, the value of those tests, and if their failure is indication of a critical state of FxA
+  - **Good**, because we can test the same, or similar business logic faster in a unit or integration test
+  - **Good**, because we also remove implicit assertions from tests. Functional tests may often test one flow, but implicitly test something along the way
```

Comment:
```suggestion
  - **Good**, because we also remove implicit assertions from tests. Functional tests may often be intended to test one flow, but implicitly test something else along the way.
```